content-src/components/DiscoveryStreamComponents/DSCard/DSCard.jsx

```diff
@@ -28,31 +29,33 @@ export class DSCard extends React.PureComponent {
 
   render() {
     return (
-      <SafeAnchor url={this.props.url} className="ds-card" onLinkClick={this.onLinkClick}>
-        <div className="img-wrapper">
-          <div className="img" style={{backgroundImage: `url(${this.props.image_src}`}} />
-        </div>
-        <div className="meta">
-          <div className="info-wrap">
-            <header className="title">{this.props.title}</header>
-            {this.props.excerpt && <p className="excerpt">{this.props.excerpt}</p>}
+      <DSLinkMenu containerClassName="ds-card" {...this.props}>
```

Comment:
Can you help me understand what we are solving with wrapping the rest of the component?

Reply:
I bring it up because of the other bugs caused by wrapping things previously, just making sure we've considered that and ensure it's not breaking things.

Reply:
[CSS on context-menu-button](https://github.com/mozilla/activity-stream/blob/master/content-src/styles/_variables.scss#L151) expects [context menu button](https://github.com/mozilla/activity-stream/pull/4818/files#diff-dc7ba3119134890bf2373097e54ada9fR44) to be at same level as anchored item (ds-card-link) and the reason for wrapping anchor here

Reply:
Ok, it doesn't seem to regress anything. So I think should be fine. Def was a rd flag though given how similar changes have caused bugs in the past.

Reply:
Same level as `ds-card-link` or parent to `ds-card-link`?

Reply:
Oh, I see, the button is returned as a sibling to `{this.props.children}`

Reply:
Couldn't you just put the DSLinkMenu as a sibling of the existing code? It definitely seems dangerous to introduce additional div nesting given the regressions from before

Reply:
Oh, I suppose a direct sibling might cause issues with the parent assuming individual children for grid reasons ?

Reply:
Yeah it makes me nervous but I'm not actually able to break it. I think Punam has covered all the cases. maybe we're fine, what do you think  @Mardak ?

Reply:
> Oh, I suppose a direct sibling might cause issues with the parent assuming individual children for grid reasons ?

yes tried to avoid nesting but  'ds-card' classname has to be at the individual item container level

Reply:
It would seem to make more sense if the components were structured as:
```jsx
card render =>
  <div className="ds-card">
    <SafeAnchor>
      … existing stuff
      … existing <ImpressionStats/>
    </SafeAnchor>
    <DSLinkMenu />
  </div>
```

So for List, DSLinkMenu doesn't need to care about `<div>` vs `<li>`. Hopefully it also makes it a little bit more clear that these visual components are creating a single container for their parents to grid align and the sibling relationship of anchor and menu.

Reply:
Actually.. there's the active state that needs to modify the "ds-card" container class list. And generally common callback behavior of the menu being needed for all the components. Probably not the time to refactor the visual components to avoid needing to reimplement in each place…

Reply:
Either that or we can use contextMenuButtonRef to access containing visual component and add active class to it, we need to explicitly remove active class when context menu hides